routing/payment_lifecycle.go

```diff
@@ -203,20 +203,41 @@ func (p *paymentLifecycle) resumePayment(ctx context.Context) ([32]byte,
 		return [32]byte{}, nil, err
 	}
 
-	// Get the payment status.
-	status := payment.GetStatus()
-
 	// exitWithErr is a helper closure that logs and returns an error.
-	exitWithErr := func(err error) ([32]byte, *route.Route, error) {
-		// Log an error with the latest payment status.
-		//
-		// NOTE: this `status` variable is reassigned in the loop
-		// below. We could also call `payment.GetStatus` here, but in a
-		// rare case when the critical log is triggered when using
-		// postgres as db backend, the `payment` could be nil, causing
-		// the payment fetching to return an error.
-		log.Errorf("Payment %v with status=%v failed: %v", p.identifier,
-			status, err)
+	exitWithErr := func(exitErr error) ([32]byte, *route.Route, error) {
+		// We fetch the latest payment status here to get the latest
+		// state of the payment.
+		payment, err := p.router.cfg.Control.FetchPayment(p.identifier)
```

Comment:
we don't need to fetch the payment again - it's already been rewritten in the loop below.

Reply:
hmm, acutally I think we need because we reload it at the beginning and then we might fail it down the road so we will not have the current payment status available.

Reply:
As long as we check for `StatusInitiated` we can be sure it's not modified, which can only happen when the async result collector is fired, which implicitly means there are inflight HTLCs, and we don't want to fail it in that case.